pkg/kubecost/service/allocation/allocationsummary.proto

```diff
@@ -0,0 +1,147 @@
+syntax = "proto3";
+
+package kubecost.service.allocation.allocationsummary;
+
+option go_package = "github.com/opencost/opencost/pkg/kubecost/service/allocation/allocationsummary";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+// Allocation Summary Interface exported by the gRPC server.
+service AllocationSummary {
+    // An RPC for allocation.
+    //
+    // Obtains the allocation summary for a given window span.
+    
+    // Obtains the AllocationSummarySets satisfy the given AllocationSummaryRequest. Results are
+    // streamed rather than returned at once.
+    rpc GetAllocationSummary (AllocationSummaryRequest) returns (stream AllocationSummaryResponse) {}
+}
+
+// AllocationSummaryRequest is the request send from client with a specify window
+// either absolute window with datetime or relative window with duration
+message AllocationSummaryRequest {
+    Window window =1;
+    repeated string aggregate =2;
+    bool accumulate=4;
+    FilterSelection filter = 5;
+    SharedSelection shared = 6;
+    IdleSelection idle = 7;
+    bool external =8;
+}
+
+// Window represents the window allocation summary is requested
+message Window{
+    oneof window{
+        AbsoluteWindow absWindow=1;
+        RelativeWindow relWindow=2;
+    }
+}
+
+// Absolute Window represents the absolute window with start and end time
+// for 
+message AbsoluteWindow {
+    google.protobuf.Timestamp start = 1;
+    google.protobuf.Timestamp end=2;
+}
+
+//Relative window represents duration for the requested data
+message RelativeWindow {
+    google.protobuf.Duration timeFrame = 1;
+}
+
+// Filter selection represents the filter on the summary data to be returned
+message FilterSelection{
+    repeated string filterClusters=1;
+    repeated string filterNodes = 2;
+    repeated string filterNamespaces = 3; 
+    repeated string filterControllerKinds = 4;
+    repeated string filterController = 5;
+    repeated string filterPods = 6;
+    repeated string filterAnnotations = 7;
+    repeated string filterLabels = 8;
+    repeated string filterServices = 9;
+}
+
+// Different selection of idle from the User Interface 
+enum IdleSelection {
```

Comment:
@michaelmdresser Can you take a look at this. How well does this line up with allocation v2 filters?

Reply:
I think v1 filters have a pretty natural proto enum representation because each field is a separate query param with comma-separated filter values (e.g. `filterNamespace=kubecost,kube-system` in HTTP API). I think the representation Alan has right now is pretty solid for that.

However, I think V2 filters (e.g. `filter=namespace:"kubecost","kube-system"+label[app]:"cost-analyzer"` in HTTP API) should stay as a string parameter because we maintain a parser for this filter language already.

It would be cool if we could enforce using either v1 filters or v2 filter but not both.